### PR TITLE
GCA command bug fix

### PR DESCRIPTION
**Pull request by v-soukum**
## Description
</br>
GCA command is not working with Reader access permission. I have added minor code to handle this exception.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK/Framework/Core/ContinuousAssurance/CAAutomation.ps1 [hunk 1]
```diff
@@ -1665,47 +1666,61 @@ class CCAutomation: AzCommandBase
...
+					$caSubs += $this.SubscriptionContext.SubscriptionId;
+					$scanobject = [CAScanModel]::new($this.SubscriptionContext.SubscriptionId, $this.LoggingOption);
+					$scanobjects += $scanobject;
+				}
 			}
 		}
+		catch 
+		{
+			$resultStatus = "Failed"
+            $failMsg = "Unable to fetch details from storage account, User do not have required permission."
```
Comment (Aboli-msft) [inline src/AzSK/Framework/Core/ContinuousAssurance/CAAutomation.ps1:1716]: Get the messages reviewed. It should not say "perform the scan"

Comment (v-soukum) [reply src/AzSK/Framework/Core/ContinuousAssurance/CAAutomation.ps1]: updated the message

File: src/AzSK/Framework/Core/ContinuousAssurance/CAAutomation.ps1 [hunk 2]
```diff
@@ -4058,17 +4076,24 @@ class CCAutomation: AzCommandBase
...
+					$date = [DateTime]::UtcNow.AddDays(-$dayCounter).ToString("yyyyMMdd")
+					$recentLogsPath = $scanLogsPrefixPattern + "AutomationLogs_" + $date
+					$recentCAScanDataBlobObject = Get-AzStorageBlob -Container $containerName -Prefix $recentLogsPath -Context $currentContext -ErrorAction SilentlyContinue
+					$dayCounter += 1
+				}
+			}
+		}
+		catch
+		{
+			throw "Unable to access AzSK scan details from storage account, The user does not seem to have the required permission. Please re-run the command after elevating owner/contributor permission."
```
Comment (TarunKrShukla) [inline src/AzSK/Framework/Core/ContinuousAssurance/CAAutomation.ps1:4096]: We should not throw exception here, better we should handle gracefully as done above.

Comment (v-soukum) [reply src/AzSK/Framework/Core/ContinuousAssurance/CAAutomation.ps1]: Changes done

**Review by TarunKrShukla — APPROVED** [review]

**Review by Aboli-msft — APPROVED** [review]

**Pull request merged**